src/main/scala/quasar/sbtdatasource/AssemblePlugin.scala

```diff
@@ -0,0 +1,205 @@
+        // are reported when any fail to download.
+        artifacts <-
+          artifactsPar.traverse[OrFileErrors, File](_.toValidatedNel).fold(
+            es => Sync[F].raiseError(new Exception(s"Failed to fetch files: ${es.foldMap(e => e.toString + "\n\n")}")),
+            Sync[F].pure(_))
+
+        // filter out coursier metadata, we only want the jars
+        // for the `classpath` field of the .plugin file
+        jarFiles = artifacts.filter(_.getName.endsWith(".jar"))
+      } yield jarFiles
+
+      // coursier prefers that we fetch metadata before fetching
+      // artifacts. we do that in parallel with copying the datasource
+      // jar to its new place, because they don't depend on one
+      // another.
+      fetchedJarFiles <- copyJar &> fetchJarFiles
+
+      _ <- Sync[F].delay(println("Artifacts fetched. Preparing to write .plugin file..."))
+
+      // the .plugin file requires all dependency jar paths
+      // to be relative to the plugins folder
+      classPath =
+        fetchedJarFiles.map(f => buildDir.relativize(f.toPath)) ::: List(relativeJarPath)
+
+      cpJson = Json.arr(classPath.map(p => Json.fromString(p.toString)) : _*)
+      mainJar = Json.fromString(relativeJarPath.toString)
+
+      // include the datasource jar and classpath into the .plugin file
+      outJson = Json.obj("mainJar" -> mainJar, "classPath" -> cpJson).spaces2
+
+      // delete an old .plugin file, write the new one
+      _ <- Sync[F] delay {
+        Files.deleteIfExists(pluginPath)
+        Files.write(pluginPath, outJson.getBytes, CREATE_NEW)
+      }
+
+      _ <- Sync[F].delay(println(".plugin file written. Zipping up tarball..."))
+
+      // equivalent to `ls $buildDir`, the files and folders we need to zip up
+      // to make a valid `plugins` folder
+      files <- Sync[F] delay {
+        Files.list(buildDir).map(buildDir.relativize(_)).toArray.mkString(" ")
+      }
+
+      // the `plugins` tarball's location
+      tarPath = dstDir.resolve(s"$dsName-$dsVersion-q$quasarVersion-explode.tar.gz")
+
+      // TODO: make this portable
+      //
+      // the command we run to finish up: zip up (-c) all of
+      // the files in our plugins folder ($files), with the
+      // plugins folder as "root" of the tarball (-C) and
+      // put the tarball into the artifacts folder.
+      cmd = s"tar -czvf $tarPath -C $buildDir/ $dsName $dsName.plugin"
+
+      // do it.
+      _ <- Sync[F].delay(Runtime.getRuntime().exec(cmd))
+
+      _ <- Sync[F].delay(println(s"Tarball written to ${tarPath}."))
+    } yield tarPath
+  }
+
+  // SBT needs `ModuleId` to declare dependencies with
+  // `libraryDependencies`, and coursier wants `Dependency`.
+  // Converting is straightforward but requires knowing the
+  // `scalaVersion`; I've hard-coded it to 2.12 here.
+  private def moduleIdToDependency(moduleId: ModuleID): Dependency =
+    Dependency(
+      Module(moduleId.organization, moduleId.name + "_2.12"),
```

Comment:
NB, I think we can actually do better than this hard-coding now that it's a plugin by taking `scalaBinaryVersion` as a parameter to `apply` and threading it through.